### PR TITLE
v5.35 Changelog

### PR DESCRIPTION
**Pull request by amyblais**
- Apps Framework (Developer Preview) section needs details on how on-prem customers can enable the feature (or a link to docs).

### CODE REVIEW

File: source/administration/changelog.md [hunk 1]
```diff
@@ -4,6 +4,125 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+     - As an example, for AWS Aurora instances, the ``QueryAbsoluteLag`` can be: ``select server_id highest_lsn_rcvd-durable_lsn as bindiff from aurora_global_db_instance_status()`` where ``server_id=<>`` and ``QueryTimeLag`` can be: ``select server_id, visibility_lag_in_msec aurora_global_db_instance_status()`` where ``server_id=<>``. For MySQL Group Replication, the absolute lag can be measured from the number of pending transactions in the applier queue: ``select member_id, count_transactions_remote_in_applier_queue FROM performance_schema.replication_group_member_stats`` where ``member_id=<>``. Overall, what query to choose is left to the administrator, and depending on the database and need, an appropriate query can be chosen.
```
Comment (amyblais) [inline source/administration/changelog.md:61]: @agnivade Is this just a release note, or does it need to be documented in important upgrade notes?

File: source/administration/changelog.md [hunk 2]
```diff
@@ -4,6 +4,125 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+ - Fixed an issue where the Team Admin’s current role was presented inconsistently in the different areas of the System Console.
+ - Fixed an issue where the **Show more** background color on long posts was broken for permalinks.
+
+### config.json
+Multiple setting options were added to ``config.json``. Below is a list of the additions and their default values on install. The settings can be modified in ``config.json``, or the System Console when available.
+
+#### Changes to Team Edition and Enterprise Edition:
+ - Under ``ServiceSettings`` in ``config.json``:
+     - Added ``EnableFileSearch`` for file search feature.
+     - Added ``EnableReliableWebSockets``.
```
Comment (amyblais) [inline source/administration/changelog.md:86]: @agnivade What would be a description for this new setting?

Comment (amyblais) [reply source/administration/changelog.md:86]: Or I guess this doesn't need to be mentioned for v5.35 yet.

File: source/administration/changelog.md (hunk 1 above)
Comment (agnivade) [reply source/administration/changelog.md:61]: Just release notes is fine. It's just a new feature - and a very well-documented one :P

File: source/administration/changelog.md (hunk 2 above)
Comment (agnivade) [reply source/administration/changelog.md:86]: Yes, let's skip this for now.

File: source/administration/changelog.md [hunk 3]
```diff
@@ -4,6 +4,124 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+   - In the v5.38 release (August 16, 2021), we will deprecate "config watcher" (the mechanism that automatically reloads the ``config.json file``), in favor of an mmctl command that will need to be run to apply configuration changes after they are made. This change will improve configuration performance and robustness.
+
+**IMPORTANT:** If you upgrade from a release earlier than v5.34, please read the other [Important Upgrade Notes](https://docs.mattermost.com/administration/important-upgrade-notes.html).
+
+### Highlights
+
+#### Apps Framework (Developer Preview)
```
Comment (amyblais) [inline source/administration/changelog.md:19]: Add a note on how to enable with env variables  https://developers.mattermost.com/integrate/apps/quick-start-go/

File: source/administration/changelog.md [hunk 4]
```diff
@@ -4,6 +4,123 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
 
 Also see [changelog in progress](https://bit.ly/2nK3cVf) for the next release.
 
+## Release v5.35 - [Feature Release](https://docs.mattermost.com/administration/release-definitions.html#feature-release)
+
+**Release Day: 2021-05-16**
+
+#### Important Upgrade Notes
+   - Due to the introduction of backend database architecture required for upcoming new features, Shared Channels and Collapsed Reply Threads, the performance of the migration process for the v5.35 release (May 16, 2021) has been noticeably affected. Depending on the size, type, and version of the database, longer than usual upgrade times should be expected. This can vary from a couple of minutes (average case) to hours (worst case, MySQL 5.x only). A moderate to significant spike in database CPU usage should also be expected during this process. [More details on the performance impact of the migration and possible mitigation strategies are provided here](https://gist.github.com/streamer45/9aee4906639a49ebde68b2f3c0f924c1).
```
Comment (cwarnermm) [inline source/administration/changelog.md:12]:
```suggestion
   - Due to the introduction of backend database architecture required for upcoming new features, including Shared Channels and Collapsed Reply Threads, the performance of the migration process for the v5.35 release (May 16, 2021) has been noticeably affected. Depending on the size, type, and version of the database, longer than usual upgrade times should be expected. This can vary from a couple of minutes (average case) to hours (worst case, MySQL 5.x only). A moderate to significant spike in database CPU usage should also be expected during this process. [More details on the performance impact of the migration and possible mitigation strategies are provided here](https://gist.github.com/streamer45/9aee4906639a49ebde68b2f3c0f924c1).
```

File: source/administration/changelog.md [hunk 5]
```diff
@@ -4,6 +4,123 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+
+**IMPORTANT:** If you upgrade from a release earlier than v5.34, please read the other [Important Upgrade Notes](https://docs.mattermost.com/administration/important-upgrade-notes.html).
+
+### Highlights
+
+#### Apps Framework (Developer Preview)
+ - Apps Framework is a new way to integrate with external tools, and allows developers to create interactive Apps in Mattermost, using any development language they're comfortable with. The new Apps work seamlessly across mobile and desktop clients. This is a developer preview and is not intended for production instances of Mattermost yet. This feature is available for self-managed customers with v5.35.0 when the Apps Framework Plugin is loaded on an instance, before then developers can use the ``cloud`` branch to get a local test environment running. The launch for the Developer Preview of the Apps Framework is scheduled for April 29th. Learn more: https://developers.mattermost.com/integrate/apps/.
```
Comment (cwarnermm) [inline source/administration/changelog.md:20]:
```suggestion
 - Apps Framework is a new way to integrate with external tools which allows developers to create interactive Apps in Mattermost, using any development language they're comfortable with. These new Apps work seamlessly across mobile and desktop clients. This is a developer preview and is not intended for production instances of Mattermost yet. This feature is available for self-managed customers with v5.35.0 when the Apps Framework Plugin is loaded on an instance, before then developers can use the ``cloud`` branch to get a local test environment running. The launch for the Developer Preview of the Apps Framework is scheduled for April 29th. Learn more: https://developers.mattermost.com/integrate/apps/.
```

Comment (cwarnermm) [reply source/administration/changelog.md:20]: I find this sentence hard to parse and understand: "This feature is available for self-managed customers with v5.35.0 when the Apps Framework Plugin is loaded on an instance, before then developers can use the ``cloud`` branch to get a local test environment running. 

Is this suggestion technically accurate?
"This feature is available for self-managed customers with v5.35.0. System Admins must load the Apps Framework Plugin on an instance before developers can use the ``cloud`` branch to get a local text environment running."

File: source/administration/changelog.md [hunk 6]
```diff
@@ -4,6 +4,123 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+
+#### Search results are returned on file search
+ - Searching in Mattermost now finds both relevant messages and files in your team's conversation history. Search will return results for attachments that match the file name or contain matching text content within supported document types. [Learn more](https://mattermost.com/blog/file-search/). To be available for mobile apps in a later release.
+
+#### Granular Access to System Console Pages (E20 Edition)
+ - Migrated Experimental, About, Reporting, Environment, Site Configuration, Authentication, Integrations, Compliance sections to their respective sub-section permissions.
```
Comment (cwarnermm) [inline source/administration/changelog.md:26]:
```suggestion
 - Migrated Experimental, About, Reporting, Environment, Site Configuration, Authentication, Integrations, and Compliance sections to their respective sub-section permissions.
```

File: source/administration/changelog.md [hunk 7]
```diff
@@ -4,6 +4,123 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+
+#### Shared Channels (Experimental, E20 Edition)
+ - Experimental support added for sharing channels between Mattermost clusters. Requires an E20 license. The Shared Channels feature is disabled by default.
+
+#### Enterprise Trial enhancements Phase 1
+ - 
+
+#### Incident Collaboration (E20 Edition)
+ - Updated prepackaged Incident Collaboration plugin with updates to Ad hoc tasks, stakeholder overview, and more.
```
Comment (cwarnermm) [inline source/administration/changelog.md:35]:
```suggestion
 - Updated prepackaged Incident Collaboration plugin with to include Ad hoc tasks, stakeholder overview, and more.
```

File: source/administration/changelog.md [hunk 8]
```diff
@@ -4,6 +4,123 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+
+#### User Interface (UI)
+ - Added support to collapse in-line images over 100px in height.
+ - Implemented maximum length validation on the status modal for custom statuses.
+ - Synchronized collapsed channel sidebar categories on the server.
+ - Empty state is no longer off-centered in the **Channel Switcher**.
+ - Ephemeral message created from call response ``markdown`` field is now posted by bot.
+ - Added improvements and fixes for the custom status feature. For example, fixed an issue where recently selected statuses were missing from the **Set a Status** confirmation screen, and updated the **Mobile Push Notifications** text in **Account Settings** to refer to user **availability** instead of **online status**.
+ - Moved the user status in the channel switcher to overlap with user avatars, and added URL 'Slug' information to channel names in the channel switcher.
+ - Added a string field to configuration for restricted domains with the key ``RestrictLinkPreviews`` and added a UI field for restricted domains under **System Console > Site Configuration > Posts**. Also expanded the logic that determines whether a post has a preview or not.
```
Comment (cwarnermm) [inline source/administration/changelog.md:47]:
```suggestion
 - Added a string field to restricted domains configuration with the key ``RestrictLinkPreviews``, and added a UI field for restricted domains under **System Console > Site Configuration > Posts**. Also expanded the logic that determines whether a post has a preview or not.
```

**Review by cwarnermm — COMMENTED** [review]
Suggestions proposed inline

File: source/administration/changelog.md [hunk 9]
```diff
@@ -4,6 +4,123 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+#### Search results are returned on file search
+ - Searching in Mattermost now finds both relevant messages and files in your team's conversation history. Search will return results for attachments that match the file name or contain matching text content within supported document types. [Learn more](https://mattermost.com/blog/file-search/). To be available for mobile apps in a later release.
+
+#### Granular Access to System Console Pages (E20 Edition)
+ - Migrated Experimental, About, Reporting, Environment, Site Configuration, Authentication, Integrations, Compliance sections to their respective sub-section permissions.
+
+#### Shared Channels (Experimental, E20 Edition)
+ - Experimental support added for sharing channels between Mattermost clusters. Requires an E20 license. The Shared Channels feature is disabled by default.
+
+#### Enterprise Trial enhancements Phase 1
```
Comment (wiersgallak) [inline source/administration/changelog.md:31]:
```suggestion
#### Enterprise Trial Enhancements 
Added banners to alert System admins when an E20 trial has started, when there are 3 days left of the trial, and when the trial is on the last day. 
```

**Review by wiersgallak — COMMENTED** [review]
Added some suggestions

**Comment by aaronrothschild** [comment]
> * Apps Framework (Developer Preview) section needs details on how on-prem customers can enable the feature (or a link to docs).

cc @justinegeffen we'll need to add the docs on running apps on-prem.  @catalintomai

File: source/administration/changelog.md [hunk 10]
```diff
@@ -4,6 +4,124 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+   - In the v5.38 release (August 16, 2021), we will deprecate "config watcher" (the mechanism that automatically reloads the ``config.json file``), in favor of an mmctl command that will need to be run to apply configuration changes after they are made. This change will improve configuration performance and robustness.
+
+**IMPORTANT:** If you upgrade from a release earlier than v5.34, please read the other [Important Upgrade Notes](https://docs.mattermost.com/administration/important-upgrade-notes.html).
+
+### Highlights
+
+#### Apps Framework (Developer Preview)
+ - Apps Framework is a new way to integrate with external tools, and allows developers to create interactive Apps in Mattermost, using any development language they're comfortable with. The new Apps work seamlessly across mobile and desktop clients. This is a developer preview and is not intended for production instances of Mattermost yet. This feature is available for self-managed customers with v5.35.0 when the Apps Framework Plugin is loaded on an instance. Learn more: https://developers.mattermost.com/integrate/apps/.
+
+#### Search results are returned on file search
```
Comment (esethna) [inline source/administration/changelog.md:23]: Search for files and document contents

Searching in Mattermost now finds both relevant messages and files in your team’s conversation history. Search will return results for attachments that match the file name or contain matching text content within supported document types. [Learn more](https://mattermost.com/blog/file-search/)

File: source/administration/changelog.md [hunk 11]
```diff
@@ -4,6 +4,124 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+
+#### Incident Collaboration (E20 Edition)
+ - Updated prepackaged Incident Collaboration plugin with to include Ad hoc tasks, stakeholder overview, and more.
+
+### Improvements
+
+#### User Interface (UI)
+ - Added support to collapse in-line images over 100px in height.
+ - Implemented maximum length validation on the status modal for custom statuses.
+ - Synchronized collapsed channel sidebar categories on the server.
```
Comment (esethna) [inline source/administration/changelog.md:43]: User benefit?

**Review by esethna — CHANGES_REQUESTED** [review]
Made a few suggestions, translating the ticket/github "language" to focus on end user benefit 

@cwarnermm when this is closer to completion propose we get an editor review to ensure we're using consistent terminology with the docs. ie "channel list" vs "channel sidebar", "avatar" vs "profile picture"

File: source/administration/changelog.md [hunk 12]
```diff
@@ -4,6 +4,124 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+     - Added ``RestrictLinkPreviews`` setting to allow disabling link previews from certain domains.
+ - Under ``FileSettings`` in ``config.json``:
+     -  Added ``ExtractContent`` and ``ArchiveRecursion`` for file search feature.
+ - Under ``ExperimentalSettings`` in ``config.json``:
+     - Added ``EnableRemoteClusterService`` for experimental Shared Channels feature.
+ - Under ``SqlSettings`` in ``config.json``:
+     - Added ``ReplicaLagSettings``. This is an array of maps which contain three keys: ``DataSource``, ``QueryAbsoluteLag``, and ``QueryTimeLag``.
+
+### Database Changes
+ - Added new column in ``ChannelMembers`` table called ``MentionCountRoot``. Please note that on installations with large amount of Channels/Users the migration can take up to a few minutes.
```
Comment (cwarnermm) [inline source/administration/changelog.md:96]:
```suggestion
 - Added new column in ``ChannelMembers`` table called ``MentionCountRoot``. Please note that the migration can take up to a few minutes on installations with large amount of Channels/Users.
```

File: source/administration/changelog.md [hunk 13]
```diff
@@ -4,6 +4,124 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+ - Added ``TotalMsgCountRoot`` to ``Channels`` table and ``MsgCountRoot`` column to ``ChannelMembers`` table. Please note that the migration on large MySQL instances can take several minutes to complete.
```
Comment (cwarnermm) [inline source/administration/changelog.md:97]:
```suggestion
 - Added ``TotalMsgCountRoot`` to ``Channels`` table and ``MsgCountRoot`` column to ``ChannelMembers`` table. Please note that the migration can take several minutes to complete on large MySQL instances.
```

**Review by cwarnermm — COMMENTED** [review]
Non-blocking suggestions provided inline

File: source/administration/changelog.md [hunk 14]
```diff
@@ -4,6 +4,124 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+ - Moved the user status in the channel switcher to overlap with user avatars, and added URL 'Slug' information to channel names in the channel switcher.
+ - Added a string field to restricted domains configuration with the key ``RestrictLinkPreviews``, and added a UI field for restricted domains under **System Console > Site Configuration > Posts**. Also expanded the logic that determines whether a post has a preview or not.
+ - Added an unread badge to the **Main Menu** icon and the **Plugin Marketplace** menu that displays until a System Admin visits the **Plugin Marketplace** for the first time.
+ - Added profile pictures to the **Direct Messages** channel list.
+ - Added channel icons for email notifications as part of email notification redesigns.
+ - Direct Messages **More...** modal is now sorted by recent conversations when the modal is opened.
+ - Removed legacy Open-Sans fonts and upgraded Open-Sans to v18.
+
+#### Administration
+ - Paused admin advisor notifications from triggering.
```
Comment (justinegeffen) [inline source/administration/changelog.md:56]:
```suggestion
 - Paused Admin Advisor notifications from triggering.
```

File: source/administration/changelog.md [hunk 15]
```diff
@@ -4,6 +4,124 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+
+### Bug Fixes
+ - Fixed link previews on a number of websites, including Reddit.
+ - Fixed an issue where SAML assigned Mattermost ``UserID`` as username if the value was invalid and did not log this.
+ - Fixed an issue where hover effects for category sorting and **Direct Messages** category limit submenus were too dark on a dark theme.
+ - Fixed an issue where users were unable to drag the vertical scroll bar on a PDF preview.
+ - Fixed an issue with animations on long posts when highlighted as a permalink.
+ - Fixed an issue where the user's nickname was not shown on channel switch.
+ - Fixed an issue where deactivated users were not marked as "Deactivated" in the channel switcher.
```
Comment (justinegeffen) [inline source/administration/changelog.md:71]:
```suggestion
 - Fixed an issue where deactivated users were not marked as **Deactivated** in the channel switcher.
```

File: source/administration/changelog.md [hunk 16]
```diff
@@ -4,6 +4,124 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+ - Fixed an issue where queries executed during the upgrade process would preemptively time out on the application side.
+ - Fixed an issue where users were unable to deactivate MFA for their accounts even if MFA was disabled on the server.
+ - Fixed an issue where user settings on API could be set if LDAP Sync was on. For LDAP and SAML users, the following fields cannot be changed via the API if the corresponding LDAP/SAML attributes have been set: first name, last name, position, nickname, email, profile picture. For OAUTH users (i.e., Gitlab, Google, Office365 and OpenID), the following fields cannot be changed via the API: first name, last name. All users who authenticate via a method other than email cannot change their username via the API.
```
Comment (justinegeffen) [inline source/administration/changelog.md:74]:
```suggestion
 - Fixed an issue where user settings on API could be set if LDAP Sync was on. For LDAP and SAML users, the following fields cannot be changed via the API if the corresponding LDAP/SAML attributes have been set: first name, last name, position, nickname, email, profile picture. For OAuth users (i.e., GitLab, Google, Office365, and OpenID), the following fields cannot be changed via the API: first name, last name. All users who authenticate via a method other than email cannot change their username via the API.
```

File: source/administration/changelog.md (hunk 12 above)
Comment (justinegeffen) [reply source/administration/changelog.md:96]: I edited this slightly, @cwarnermm. Hope that's ok. :)

File: source/administration/changelog.md [hunk 17]
```diff
@@ -4,6 +4,124 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+ - Under ``SqlSettings`` in ``config.json``:
+     - Added ``ReplicaLagSettings``. This is an array of maps which contain three keys: ``DataSource``, ``QueryAbsoluteLag``, and ``QueryTimeLag``.
+
+### Database Changes
+ - Added new column in ``ChannelMembers`` table called ``MentionCountRoot``. Please note that on installations with large amount of Channels/Users the migration can take up to a few minutes.
+ - Added ``TotalMsgCountRoot`` to ``Channels`` table and ``MsgCountRoot`` column to ``ChannelMembers`` table. Please note that the migration on large MySQL instances can take several minutes to complete.
+
+### API Changes
+ - Added ``/teams/{team_id}/files/search`` API endpoint for files search.
+ - For LDAP and SAML users, the following fields cannot be changed via the API if the corresponding LDAP/SAML attributes have been set: first name, last name, position, nickname, email, profile picture. For OAUTH users (i.e. Gitlab, Google, Office365 and OpenID), the following fields cannot be changed via the API: first name, last name. All users who authenticate via a method other than email cannot change their username via the API.
```
Comment (justinegeffen) [inline source/administration/changelog.md:101]:
```suggestion
 - For LDAP and SAML users, the following fields cannot be changed via the API if the corresponding LDAP/SAML attributes have been set: first name, last name, position, nickname, email, profile picture. For OAuth users (i.e., GitLab, Google, Office365, and OpenID), the following fields cannot be changed via the API: first name, last name. All users who authenticate via a method other than email cannot change their username via the API.
```

File: source/administration/changelog.md [hunk 18]
```diff
@@ -4,6 +4,124 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+
+### Go Version
+ - v5.35 is built with Go ``1.15.5``.
+
+### Open Source Components
+ - Added ``country-list``, ``form-data``, ``gfycat-sdk``, ``redux-thunk``, ``rudder-sdk-js``, ``serialize-error`` and ``shallow-equals``, and removed ``mattermost-redux`` from https://github.com/mattermost/mattermost-webapp.
+
+### Known Issues
+ - Pinned posts are no longer highlighted.
+ - ``Config.json`` can reset when running the command ``systemctl restart mattermost``, and when running any commands that write to the config (e.g. ``config`` or ``plugin``) [MM-33752](https://mattermost.atlassian.net/browse/MM-33752), [MM-32390](https://mattermost.atlassian.net/browse/MM-32390).
```
Comment (justinegeffen) [inline source/administration/changelog.md:111]:
```suggestion
 - ``config.json`` can reset when running the command ``systemctl restart mattermost``, and when running any commands that write to the config (e.g. ``config`` or ``plugin``) [MM-33752](https://mattermost.atlassian.net/browse/MM-33752), [MM-32390](https://mattermost.atlassian.net/browse/MM-32390).
```
